packages/dds/map/src/test/mocha/map.spec.ts

```diff
@@ -773,6 +777,133 @@ describe("Map", () => {
+					map.getAllAttribution()?.size,
+					0,
+					"do not have correct number of attribution",
+				);
+				assert.equal(
+					summaryContent,
+					`{"blobs":[],"content":{"first":{"type":"Plain","value":"second"},"third":{"type":"Plain","value":"fourth"},"fifth":{"type":"Plain","value":"sixth"}}}`,
+				);
+			});
+		});
+
+		describe("Connected State", () => {
+			let containerRuntimeFactory: MockContainerRuntimeFactory;
+			let map1: SharedMap;
+			let map2: SharedMap;
+
+			beforeEach(async () => {
+				const options: IMapOptions = { attribution: { track: true } };
+				containerRuntimeFactory = new MockContainerRuntimeFactory();
+				// Create the first map
+				map1 = createConnectedMap("map1", containerRuntimeFactory, options);
+				// Create and connect a second map
+				map2 = createConnectedMap("map2", containerRuntimeFactory, options);
+			});
+
+			it("Can generate proper attribution in connected state", () => {
+				map1.set("first", "value1");
+				map1.set("second", "value2");
+				map2.set("first", "value3");
+
+				containerRuntimeFactory.processSomeMessages(1);
+
+				assert.notEqual(
+					map1.getAttribution("first"),
+					undefined,
+					"map1 did not generate correct attribution",
+				);
+
+				containerRuntimeFactory.processAllMessages();
+
+				assert.deepEqual(
+					map1.getAllAttribution(),
+					map2.getAllAttribution(),
+					"map1 and map2 do not have consistent attribution",
+				);
+
+				map1.clear();
+				containerRuntimeFactory.processAllMessages();
+				assert.equal(
+					map1.getAllAttribution()?.size,
+					0,
+					"the attribution of map1 was not cleared",
+				);
+				assert.equal(
+					map2.getAllAttribution()?.size,
+					0,
+					"the attribution of map2 was not cleared",
+				);
+			});
+		});
+
+		describe("Summarization", () => {
+			let containerRuntimeFactory: MockContainerRuntimeFactory;
+			let map1: SharedMap;
+			let map2: SharedMap;
+
+			beforeEach(async () => {
+				const options: IMapOptions = { attribution: { track: true } };
+				containerRuntimeFactory = new MockContainerRuntimeFactory();
+				// Create the first map
+				map1 = createConnectedMap("map1", containerRuntimeFactory, options);
+				// Create and connect a second map
+				map2 = createConnectedMap("map2", containerRuntimeFactory, options);
+			});
+
+			it("Can summarize/load the map correctly with attribution", async () => {
+				map1.set("first", "value1");
+				map1.set("second", "value2");
+
+				containerRuntimeFactory.processAllMessages();
+
+				const summaryContent = (map1.getAttachSummary().summary.tree.header as ISummaryBlob)
+					.content;
+
+				assert.equal(
+					map1.getAllAttribution()?.size,
+					2,
+					"do not have correct number of attribution",
+				);
+				assert.equal(
+					summaryContent,
+					`{"blobs":[],"content":{"first":{"type":"Plain","value":"value1","attribution":{"type":"op","seq":1}},"second":{"type":"Plain","value":"value2","attribution":{"type":"op","seq":2}}}}`,
```

Comment:
since we're already adding bespoke code for op-based attribution in merge-tree, i think we might as well do it here too. That would look like `"attribution":1` rather than `"attribution":{"type":"op","seq":1}` here.

Reply:
later down the line it might be worth putting serialization of attribution keys through a common codepath. there's not a great place for such a codepath right now though (DDS stuff can't depend on @fluidframework/attributor due to layering, and we don't want to put functionality in e.g. definitions packages).